### PR TITLE
update schema when a workflow field has a valueProviderName field

### PR DESCRIPTION
**Pull request by dagda1**
If  a workflow contains a `valueProviderName` field then call the `/workflowDefinitions` endpoint.

An example of a response is

```json
[
  {
     key: 'WORKFLOW_SELECT_SAMPLE',
     value: 'option2',
     workName: 'complexWorkFlow',
      propertyPath: 'complexWorkFlow',
   }
]
```

Which causes a schema update and a UI re-render:

https://user-images.githubusercontent.com/118328/235970409-b72cbb44-dcbd-4ac1-a3d0-22e789fe9633.mp4

### CODE REVIEW

**Review by RichardW98 — APPROVED** [review]

File: plugins/parodos/src/components/Form/widgets/SelectWidget.tsx [hunk 1]
```diff
@@ -1,7 +1,68 @@
...
+        await updateSchema([
+          {
+            key: props.schema.title,
+            value,
+            workName: workflowDefinitionName,
+          },
+        ]);
+      }
+
+      onChange(value);
```
Comment (wKich) [inline plugins/parodos/src/components/Form/widgets/SelectWidget.tsx:41]: Possible race-condition

**Review by wKich — APPROVED** [review]

File: plugins/parodos/src/components/Form/widgets/SelectWidget.tsx (hunk 1 above)
Comment (RichardW98) [reply plugins/parodos/src/components/Form/widgets/SelectWidget.tsx:41]: maybe we can disable all the fields in the form and enable them again after the onChange() call?

Comment (dagda1) [reply plugins/parodos/src/components/Form/widgets/SelectWidget.tsx:41]: The select is disabled by by the loading flag, see below.  Why would we need to disable the whole form

Comment (RichardW98) [reply plugins/parodos/src/components/Form/widgets/SelectWidget.tsx:41]: sorry I thought the race-condition is between multiple fields in the same form.  @wKich  please help explain a bit more, Thank you

**Pull request merged**